### PR TITLE
wasi: adds Zig stdlib test case for 

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Comment by mathetake** [comment]
Zig's coming-along wasi-libc has been updated recently(https://github.com/ziglang/zig/commit/da9c530d9937abd1c17b9eeabec199076428a177), and that comes with this commit(https://github.com/WebAssembly/wasi-libc/commit/60f221a400e36ba59d99c191a86e978117320f0a#diff-c61ee8b562c86d6118d199772bc65dedb72199b4e783f3119039cbb75ccac01c) in WebAssembly/wasi-libc, and it seems that is the root cause

**Comment by mathetake** [comment]
seesms like there's some glitch in the wasi-libc-update in the Zig

**Comment by mathetake** [comment]
will dig into later

**Comment by mathetake** [comment]
feels like https://github.com/ziglang/zig/commit/af4361f57af388238a075b0c8ef5b34e75b73787 this is the fix

**Comment by mathetake** [comment]
still failing on master - https://github.com/ziglang/zig/issues/13762 submitted to the ziglang/zig

**Comment by mathetake** [comment]
still no luck 😢

**Comment by mathetake** [comment]
started working now!

File: Makefile [hunk 1]
```diff
@@ -51,14 +51,12 @@ test.examples:
 build.examples.as:
 	@cd ./imports/assemblyscript/example/testdata && npm install && npm run build
 
-# Use -fstage1 to avoid bugs in the new compiler
-# https://github.com/ziglang/zig/wiki/Self-Hosted-Compiler-Upgrade-Guide#is-it-time-to-upgrade
 %.wasm: %.zig
-	@(cd $(@D); zig build -fstage1 -Drelease-small=true)
```
Comment (codefromthecrypt) [inline Makefile:57]: bridges burnt all ahead full!

File: imports/wasi_snapshot_preview1/testdata/zig/build.zig [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const std = @import("std");
+const CrossTarget = std.zig.CrossTarget;
+
+pub fn build(b: *std.build.Builder) void {
+    // Standard release options allow the person running `zig build` to select
+    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall.
+    const mode = b.standardReleaseOptions();
+
+    const exe = b.addExecutable("wasi", "wasi.zig");
+    exe.linkLibC(); // Temporarily force libc usage due to the std.os.isatty bug.
```
Comment (mathetake) [inline imports/wasi_snapshot_preview1/testdata/zig/build.zig:10]: ref https://github.com/ziglang/zig/pull/13813

**Review by codefromthecrypt — APPROVED** [review]
coolness

**Pull request merged**